### ISSUE TITLE
Docs need to be updated with the new withEndpoints method for service discovery

### ISSUE THREAD

**Issue by PhilStollery** [dotnet/docs-aspire#391]
**Issue description**

The current page: https://learn.microsoft.com/en-us/dotnet/aspire/service-discovery/overview#named-endpoints-in-net-aspire

Uses the replaced **.WithServiceBinding** instead of the current **.WithEndpoint** method.

**Target framework**

- [ x ] .NET Core 8
- [ ] .NET Framework
- [ ] .NET Standard 8

<details>
<summary><strong>dotnet --info output</strong></summary>

```console
.NET SDK:
 Version:           8.0.200
 Commit:            438cab6a9d
 Workload version:  8.0.200-manifests.5638171e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.200\

.NET workloads installed:
 [aspire]
   Installation Source: VS 17.10.34607.79
   Manifest Version:    8.0.0-preview.3.24105.21/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.0.0-preview.3.24105.21\WorkloadManifest.json
   Install Type:              Msi


Host:
  Version:      8.0.2
  Architecture: x64
  Commit:       1381d5ebd2

.NET SDKs installed:
  6.0.419 [C:\Program Files\dotnet\sdk]
  8.0.102 [C:\Program Files\dotnet\sdk]
  8.0.200 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```
</details>

**Comment by timdeschryver** [comment]
I came here to create a similar issue for the networking page: https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/networking-overview (content at https://github.com/dotnet/docs-aspire/blob/main/docs/fundamentals/networking-overview.md)

The snippets referenced on this page are already updated in #389.

**Comment by IEvangelist** [comment]
Closed by #393
